### PR TITLE
added SearchView Support

### PR DESCRIPTION
**Pull request by keineantwort**
Support for androids `SearchView` with handling the query

### CODE REVIEW

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/KSearchView.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.agoda.kakao.common.builders.ViewBuilder
+import com.agoda.kakao.common.views.KBaseView
+import org.hamcrest.Matcher
+
+class KSearchView : KBaseView<KSearchView>, SearchViewActions, SearchViewAssertions {
+
+	constructor(function: ViewBuilder.() -> Unit) : super(function)
+	constructor(parent: Matcher<View>, function: ViewBuilder.() -> Unit) : super(parent, function)
+	constructor(parent: DataInteraction, function: ViewBuilder.() -> Unit) : super(parent, function)
+}
```
Comment (Unlimity) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/KSearchView.kt:14]: Please add newline at the end of the file

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                description.appendText("with SearchView query: $query, but was: $foundQuery")
+            }
+
+            override fun matchesSafely(item: SearchView): Boolean {
+                foundQuery = item.query.toString()
+                return foundQuery == query
+            }
+        }
+    }
+}
```
Comment (Unlimity) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt:47]: Please add newline at the end of the file

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/KSearchView.kt (hunk 1 above)
Comment (keineantwort) [reply kakao/src/main/kotlin/com/agoda/kakao/searchview/KSearchView.kt]: done

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt (hunk 2 above)
Comment (keineantwort) [reply kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt]: done

File: sample/src/main/res/layout/activity_test.xml [hunk 3]
```diff
@@ -178,8 +183,8 @@
       android:layout_width="wrap_content"
       android:layout_height="match_parent"
       android:layout_gravity="start"
-      android:fitsSystemWindows="true"
-      android:background="@color/background_color">
+      android:background="@color/background_color"
+      android:fitsSystemWindows="true">
```
Comment (Vacxe) [inline sample/src/main/res/layout/activity_test.xml:187]: Does reordering make any sense?

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/KSearchView.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.agoda.kakao.searchview
+
+import android.view.View
+import androidx.test.espresso.DataInteraction
+import com.agoda.kakao.common.builders.ViewBuilder
+import com.agoda.kakao.common.views.KBaseView
+import org.hamcrest.Matcher
+
+class KSearchView : KBaseView<KSearchView>, SearchViewActions, SearchViewAssertions {
+
```
Comment (Vacxe) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/KSearchView.kt:10]: Remove empty line plz

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewActions.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers
+import com.agoda.kakao.common.actions.BaseActions
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+
+interface SearchViewActions : BaseActions {
+
+	fun typeQuery(query: String, append: Boolean = false) {
```
Comment (Vacxe) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewActions.kt:14]: Providing `boolean` as argument to function actually not a clear way. _(Clean code - Robert Martin )_

Could you please separate logic for two different functions for single responsibility purpose such as

```
fun typeQuery
fun appendQuery
```

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import androidx.annotation.StringRes
+import androidx.test.espresso.matcher.BoundedMatcher
+import com.agoda.kakao.common.assertions.BaseAssertions
+import com.agoda.kakao.common.utilities.getResourceString
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+
+@RequiresApi(Build.VERSION_CODES.JELLY_BEAN)
+interface SearchViewAssertions : BaseAssertions {
+
```
Comment (Vacxe) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt:16]: Remove empty line

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    fun hasHint(hint: String): Matcher<View> {
+        return object : BoundedMatcher<View, SearchView>(SearchView::class.java), Matcher<View> {
```
Comment (Vacxe) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt:18]: Can be single lined

File: kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            override fun matchesSafely(item: SearchView): Boolean {
+                return item.queryHint?.toString() == hint
+            }
+        }
+    }
+
+    fun hasHint(@StringRes queryHintId: Int): Matcher<View> = hasHint(getResourceString(queryHintId))
+
+    fun hasQuery(query: String): Matcher<View> {
+        return object : BoundedMatcher<View, SearchView>(SearchView::class.java), Matcher<View> {
```
Comment (Vacxe) [inline kakao/src/main/kotlin/com/agoda/kakao/searchview/SearchViewAssertions.kt:34]: Can be single lined

**Review by Vacxe — CHANGES_REQUESTED** [review]
Thanks for the contribution! Sorry for delay because of Christmas!

File: sample/src/main/res/layout/activity_test.xml (hunk 3 above)
Comment (keineantwort) [reply sample/src/main/res/layout/activity_test.xml:187]: It's just the default Code-Style in Android Studio.

**Review by Unlimity — APPROVED** [review]

File: .idea/codeStyles/Project.xml [hunk 9]
```diff
@@ -13,6 +13,22 @@
...
+      </option>
+      <option name="PACKAGES_IMPORT_LAYOUT">
+        <value>
+          <package name="" alias="false" withSubpackages="true" />
+          <package name="java" alias="false" withSubpackages="true" />
+          <package name="javax" alias="false" withSubpackages="true" />
+          <package name="kotlin" alias="false" withSubpackages="true" />
+          <package name="" alias="true" withSubpackages="true" />
+        </value>
+      </option>
```
Comment (Vacxe) [inline .idea/codeStyles/Project.xml:31]: Please remove this changes

**Comment by Vacxe** [comment]
@keineantwort please refuse code style changes in `Project.xml` and we can merge the changes. Thanks!

File: .idea/codeStyles/Project.xml (hunk 9 above)
Comment (keineantwort) [reply .idea/codeStyles/Project.xml]: Done. Sorry for that. Was an automatic change by Android Studio.

**Review by Vacxe — APPROVED** [review]

**Pull request merged**